### ISSUE TITLE
First alignment to initialized graph falsely assigns mismatch to final node

### ISSUE THREAD

**Issue by rlorigro** [ljdursi/poapy#3]
When (globally) aligning a set of identical or very similar sequences, the first aligned sequence is given a false mismatch in the final character, but all subsequent alignments correctly assign the last character as a match.

 Example output:

> Sequences before alignment:
> 0 TATACCGGCG
> 1 TATACCGGCG
> 2 TATACCGGCG
> 3 TATACCGGCG
> 4 TATACCGGCG
> 5 TATACCGGCG
> 
> Aligned sequences:
> 0               TATACCGGCG-
> 1               TATACCGGC-G
> 2               TATACCGGCG-
> 3               TATACCGGCG-
> 4               TATACCGGCG-
> 5               TATACCGGCG-
> Consensus0      TATACCGGCG-

Code to reproduce this issue:

```
import random
from poapy import poagraph
from poapy import seqgraphalignment


def generate_identical_sequences(n_sequences, length, character_pool):
    sequence = generate_sequence(length=length, character_pool=character_pool)

    sequences = [sequence]*n_sequences

    return sequences


def generate_sequence(length, character_pool):
    sequence = list()

    for i in range(length):
        sequence.append(random.choice(character_pool))

    sequence_string = ''.join(sequence)

    return sequence_string


if __name__ == "__main__":
    character_pool = ["A","C","G","T"]

    sequences = generate_identical_sequences(n_sequences=6, length=10, character_pool=character_pool)

    print("SEQUENCES BEFORE ALIGNMENT:")
    for s, sequence in enumerate(sequences):
        print(s, sequence)

    init_sequence = sequences[0]
    init_label = "0"

    graph = poagraph.POAGraph(init_sequence, init_label)

    for i in range(1, len(sequences)):
        sequence = sequences[i]
        read_id = str(i)

        alignment = seqgraphalignment.SeqGraphAlignment(sequence, graph,
                                                        fastMethod=True,
                                                        globalAlign=True,
                                                        matchscore=1,
                                                        mismatchscore=-1,
                                                        gapscore=-2)

        graph.incorporateSeqAlignment(alignment, sequence, str(i))

    alignments = graph.generateAlignmentStrings()

    print("\nALIGNED SEQUENCES:")
    for read_id, alignstring in alignments:
        print("{0:15s} {1:s}".format(read_id, alignstring))
```

**Comment by ljdursi** [comment]
Nice bug report, thanks!

**Comment by ljdursi** [comment]
Thanks for the very clear bug report.  I've fixed it, and started a test suite which currently just has your test case (for a variety of number of sequences, local/global alignments, and the simple/fast methods).

**Comment by ljdursi** [comment]
Fixed by #5
